### PR TITLE
Add jdsports_gb (400 sports goods stores)

### PR DESCRIPTION
**Pull request by mjoe999**

### CODE REVIEW

File: locations/spiders/jdsports_gb.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    ]
+    download_delay = 0.2
+
+    def parse_store(self, response):
+        if "-soon" in response.url:
+            self.logger.warning("ignoring store opening soon %s", response.url)
+        else:
+            item = LinkedDataParser.parse(response, "Store")
+            if item:
+                item["ref"] = response.url
```
Comment (iandees) [inline locations/spiders/jdsports_gb.py:27]: Just use the unique part of the URL:

```suggestion
                item["ref"] = response.url.strip('/').split('/')[-1]
```

File: locations/spiders/jdsports_gb.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    start_urls = ["https://www.jdsports.co.uk/store-locator/all-stores/"]
+    rules = [
+        Rule(
+            LinkExtractor(allow="store-locator/"), callback="parse_store", follow=False
+        )
+    ]
+    download_delay = 0.2
+
+    def parse_store(self, response):
+        if "-soon" in response.url:
```
Comment (Cj-Malone) [inline locations/spiders/jdsports_gb.py:22]: Can this be done as a 'deny' in the LinkExtractor? That way we'd skip the http request.

File: locations/spiders/jdsports_gb.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            self.logger.warning("ignoring store opening soon %s", response.url)
+        else:
+            item = LinkedDataParser.parse(response, "Store")
+            if item:
+                item["ref"] = response.url.strip("/").split("/")[-1]
+                yield item
```
Comment (Cj-Malone) [inline locations/spiders/jdsports_gb.py:28]: Unimportant, but since this function only ever returns 1 item, this doesn't need to be a 'yield'

**Review by Cj-Malone — COMMENTED** [review]
For future us, the ld data here gives us `"country": "UNITED KINGDOM"`, we should probably ensure `country` is an ISO country code. In this case "GB".

File: locations/spiders/jdsports_gb.py (hunk 2 above)
Comment (mjoe999) [reply locations/spiders/jdsports_gb.py:22]: There's just 2 of them. It's clearer in code I think.

**Review by Cj-Malone — APPROVED** [review]

**Pull request merged**